xds/src/main/java/io/grpc/xds/internal/certprovider/CertificateProvider.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds.internal.certprovider;
+
+import java.security.PrivateKey;
+import java.security.cert.X509Certificate;
+import java.util.List;
+
+/**
+ * A plug-in that provides certificates required by the xDS security component and created
+ * using the certificate-provider config from the xDS server.
+ *
+ * <p>We may move this out of the internal package and make this an official API in the future.
```

Comment:
This is meant to be a regular comment and not a TODO comment because this is not a short-term workaround but rather a long term goal documented in the comment.